.github/sync_labels.py

```diff
@@ -0,0 +1,348 @@
+            self.edit(label, '--remove-label')
+            info('Remove label from %s: %s' % (self._issue, label))
+
+    def on_label_remove(self, label):
+        """
+        Check if the given label belongs to a selection list. If so, reject
+        the removement in case it represents the stat of a PR.  In all other
+        cases add the successor of the label except if there is none or there
+        exists another label of the corresponding list.
+        """
+        sel_list = selection_list(label)
+        if not sel_list:
+            return
+
+        if label in self.get_labels():
+            # this is possible if two labels of the same selection list
+            # have been removed in one step (via multiple selection in the
+            # pull down menue). In this case `label` has been added
+            # on the `on_label_remove` of the first of the two labels.
+            partn = self.active_partners(label)
+            if not partn:
+                self.add_comment('Label *%s* can not be removed (last one of list)!' % label)
+            else:
+                self.on_label_add(partn[0])
+            return
+
+        item = sel_list(label)
+        if sel_list is State and self._pr:
+            if item in [State.positive_review, State.needs_work]:
+                self.add_comment('Label *%s* can not be removed. Please use the corresponding functionality of GitHub' % label)
+                self.select_label(label)
+                return
+
+        if not self.active_partners(label):
+            succ = sel_list(label).succ()
+            if succ:
+                # add the next weaker label
+                self.select_label(succ.value)
+            else:
+                self.add_comment('Label *%s* can not be removed since there is no replacement' % label)
+                self.select_label(label)
+            
+
+###############################################################################
+# Main
+###############################################################################
+cmdline_args = sys.argv[1:]
+
+getLogger().setLevel(INFO)
+info('cmdline_args (%s) %s' % (len(cmdline_args), cmdline_args))
+
+if len(cmdline_args) < 4:
+    print('Need 4 arguments: action, url, label, rev_state' )
+    exit
+else:
+    action, url, label, rev_state = cmdline_args
+
+info('action: %s' % action)
+info('url: %s' % url)
+info('label: %s' % label)
+info('rev_state: %s' % rev_state)
+
+gh = GhLabelSynchronizer(url)
+
+if action == 'opened':
+    gh.add_default_label(Priority.major.value)
+    gh.add_default_label(IssueType.enhancement.value)
+    if gh.is_pull_request():
+        if not gh.is_draft():
+            gh.add_default_label(State.needs_review.value)
+
+if action == 'reopened':
+    if gh.is_pull_request():
+        if not gh.is_draft():
+            gh.add_default_label(State.needs_review.value)
+
+if action == 'closed':
+    for lab in State:
+        gh.remove_label(lab.value)
```

Comment:
Not sure, but maybe it's a good idea to keep the status label when an issue or PR is closed. For example, an PR closed with "positive review" means it is merged, while a PR closed with "needs work" is closed because it got stale.

Reply:
In general, I think that labels, which was added by a human, should not be removed without human intervention.

Reply:
> Not sure, but maybe it's a good idea to keep the status label when an issue or PR is closed. For example, an PR closed with "positive review" means it is merged, while a PR closed with "needs work" is closed because it got stale.

My intention is to have the meaning of the state-labels as close to their meaning in Trac as possible, since their names imply that this would be the case. So, if a Trac ticket is closed it is not in `positive review` any more. The information that you want to preserve can be recovered from the progress of state changes in the ticket and the same is true for GitHub PR.

BTW: Is it possible to reopen a merged PR by someone else than the release manager? Anyway, aren't there cases where we have to remove the state labels on `reopened`-event if we don't do it on `closed`?

Reply:
> In general, I think that labels, which was added by a human, should not be removed without human intervention.

I'm less optimistic with respect to human discipline. At least if the person who should remove the label is different from the one who added it, it will not happen in many cases.

Reply:
According to Kwankyu's recent comment I let this as is, but add the `reopened` trigger.